### PR TITLE
Add subcodes to Elasticsearch and Internal engine service

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

Jira : [Subtask](https://jira.kaliop.net/browse/KZL-1189) and [Subtask](https://jira.kaliop.net/browse/KZL-1191) from this [story](https://jira.kaliop.net/browse/KZL-1188)

Add subcodes to ElasticSearch and Internal engine service
Add/adapt unit test.

### How should this be manually tested?

unit/functional tests

### Boyscout

80 columns

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [index.js#L101](https://github.com/kuzzleio/kuzzle/blob/0e510dfc4b12b4547f135ebf478b57ba41b0c7e9/lib/services/internalEngine/index.js#L101): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [index.js#L101](https://github.com/kuzzleio/kuzzle/blob/9eb8e0b3affe8e87fd932642a589cf9a87f4d329/lib/services/internalEngine/index.js#L101): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/services/elasticsearch.js [hunk 1]
```diff
@@ -1244,16 +1415,29 @@ class ElasticSearch extends Service {
...
+                    _type: type,
+                    _id: document._id}
+                });
               esRequest.body.push(document._source);
               toImport.push(document);
             } else {
-              extracted.rejected.push({document, reason: 'cannot replace a non-existing document (use mCreateOrReplace if you need to create non-existing documents)'});
+              extracted.rejected.push(
+                {document,
+                  reason: 'cannot replace a non-existing document ' +
```
Comment (kuzzle) [inline lib/services/elasticsearch.js]: MINOR Convert this concatenation to the use of a template. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3512)

File: lib/services/elasticsearch.js [hunk 2]
```diff
@@ -74,7 +69,11 @@ class ElasticSearch extends Service {
     }
 
     if (process.env.NODE_ENV === 'production' && this.config.dynamic === 'true') {
-      this.kuzzle.log.warn('Your dynamic mapping policy is set to \'true\' for new fields.\nElasticsearch will try to automatically infer mapping for new fields, and those cannot be changed afterward.\nSee the "services.db.dynamic" option in the kuzzlerc configuration file to change this value.'
+      this.kuzzle.log.warn('Your dynamic mapping policy is set to \'true\' ' +
```
Comment (kuzzle) [inline lib/services/elasticsearch.js]: MINOR Convert this concatenation to the use of a template. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3512)

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [index.js#L101](https://github.com/kuzzleio/kuzzle/blob/1149ec87764db6424a842ab4ad90793dc89b5d5b/lib/services/internalEngine/index.js#L101): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [index.js#L101](https://github.com/kuzzleio/kuzzle/blob/2f59bbc047d1ad5ce7cf597a3f757c133ba99f2f/lib/services/internalEngine/index.js#L101): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [index.js#L101](https://github.com/kuzzleio/kuzzle/blob/abe09a39d3dc1e91ec63275c5bd155efd6d555c0/lib/services/internalEngine/index.js#L101): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [index.js#L101](https://github.com/kuzzleio/kuzzle/blob/5f8837ffc8175d251a9faf9b8f89d458e8fc3c4c/lib/services/internalEngine/index.js#L101): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/services/service.js [hunk 3]
```diff
@@ -38,6 +39,22 @@ class Service {
     return this.kuzzle.internalEngine
       .createOrReplace('services', this.settings.service, this.settings || {});
   }
+
+  throw(...args) {
+
```
Comment (alexandrebouthinon) [inline lib/services/service.js]:
```suggestion
```

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [index.js#L101](https://github.com/kuzzleio/kuzzle/blob/9ee6b6bd83b022778a9b79bd86acd1f23f59e3ca/lib/services/internalEngine/index.js#L101): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**